src/appshell/view/dockwindow/dockwindow.cpp

```diff
@@ -436,20 +433,26 @@ void DockWindow::restorePageState(const QString& pageName)
 {
     TRACEFUNC;
 
-    QByteArray state = configuration()->pageState(pageName);
-    if (state.isEmpty()) {
-        return;
-    }
-
     /// NOTE: Do not restore geometry
-    KDDockWidgets::RestoreOption option = KDDockWidgets::RestoreOption::RestoreOption_RelativeToMainWindow;
-
-    KDDockWidgets::LayoutSaver layoutSaver(option);
-    if (!layoutSaver.restoreLayout(state)) {
+    bool ok = restoreLayout(configuration()->pageState(pageName), KDDockWidgets::RestoreOption::RestoreOption_RelativeToMainWindow);
+    if (!ok) {
         LOGE() << "Could not restore the state of " << pageName << "!";
     }
 }
 
+bool DockWindow::restoreLayout(const QByteArray& layout, KDDockWidgets::RestoreOptions)
+{
+    if (layout.isEmpty()) {
+        return true;
+    }
+
+    //! TODO: restoring of layout is temporary disabled because it causes many troubles for developers
+    //KDDockWidgets::LayoutSaver layoutSaver(option);
+    //return layoutSaver.restoreLayout(state);
```

Comment:
Maybe we need to add a message to the log so that we don't forget and it reminds us of this moment.

Reply:
Do you mean the NOT_IMPLEMENTED macro?

Reply:
I think like:
```
LOGI() << "TODO: restoring of layout is temporary disabled because it troubles ";
```

Reply:
fixed